### PR TITLE
Add option for additional logo in footer

### LINKED ISSUE #3134: Add option for additional logo in footer
Here's a formatted GitHub issue for your footer component updates:

**Title**: Update Footer Component for Multiple Logos

**Description**: Our footer component needs the following updates:

1. **Support Multiple Logos**:
    - Allow for two logos instead of just one.
    - Enable setting links for each logo.

2. **Default Layout**:
    - The default layout should have the logos positioned next to each other.

3. **Alternative Layout**:
    - Add an option to display the logos on opposite sides of the footer.

4. **Consistency with Header Component**:
    - Align and standardize this work with the current implementation used in the header component.
    - Follow the `multipleLogo` implementation:
        - `MultipleLogo` overrides the default ONS logo if SVGs or images are provided.
        - You can also pass in "ONS logo" to use the default ONS logo.

5. **Focus and Hover States**:
    - Ensure that the focus and hover states do not create an extra underline.
    - The highlight color should only wrap the SVG and not its margin.

**Tasks**:
- [x] Modify the footer component to support two logos.
- [x] Implement link settings for each logo.
- [x] Set default layout to have logos next to each other.
- [x] Add an option for logos to be displayed on opposite sides of the footer.
- [x] Standardize implementation with the header component's `multipleLogo` feature.
- [x] Ensure the focus and hover states meet the specified requirements.

**Additional Information**:
- Follow the existing implementation for multiple logos in the header component.
- Ensure consistent behavior and appearance across the application.

Similar work was done for the header component, check out that PR https://github.com/ONSdigital/design-system/pull/2994

### PR DESCRIPTION
**Pull request by precious-onyenaucheya-ons**
### What is the context of this PR?

Fixes #3134 
This PR updates the footer component to support an additional logo, allowing for two logos instead of just one. It introduces link param for each logo and offers an alternative layout option to display the logos on opposite sides of the footer. The default layout positions the logos next to each other. Additionally, the implementation is aligned with the header component's multipleLogo feature to ensure consistency across the application. The focus and hover states are also adjusted to avoid extra underlines and to ensure the highlight color wraps only the SVG.

_BREAKING CHANGE_
Services using the poweredBy param with the footer component will now need to update that param to footerLogo object in their config.

### How to review this PR

- Spin up the website and head to the 'footer' component, the example-footer-with-extra-poweredby-logo on the page is the one to look at.
- Check for multiple logos: Verify that the footer displays two logos as expected, with appropriate links set for each logo.
- Review layout options: Confirm that the default layout positions the logos next to each other and test the alternative layout option where the logos are displayed on opposite sides of the footer.
- Test focus and hover states: Ensure that the focus and hover states do not create an extra underline and that the highlight color wraps only the SVG, not its margin.
- run yarn test --testNamePattern="macro: footer" & yarn test-visual

### Checklist

This needs to be completed by the person raising the PR.

- [x] I have selected the correct Assignee
- [x] I have linked the correct Issue

### CODE REVIEW

**Review by ons-ds-team — COMMENTED** [review]
Could you please create another example for the oppositeDisplay option and also add a test for this specific parameter?

**Review by rmccar — CHANGES_REQUESTED** [review]
Screenshot 2024-06-13 at 09 39 19
Some of the footer examples seem to have extra padding now

**Comment by balibirchlee** [comment]
I thought that we were wanting to rename 'poweredby' - and follow the naming convention of the header section with the logo1 logo2 rather than 'additionalPoweredByLogo'. Given that the functionality has come from the team, having it as a breaking change / incorporating it into plans for deprecation makes more sense to me, as there's no external pressure for it to be merged asap - I worry about creating more tech debt

**Comment by precious-onyenaucheya-ons** [comment]
Previously, a margin-bottom was set for the default ONS logo if no powered-by logo was provided. However, if a powered-by logo was included, there was no margin-bottom. The difference in the snapshot is due to my adjustment: I have added the margin-bottom to be consistently applied in both scenarios.

File: src/components/footer/_macro-options.md
Comment (rmccar) [inline src/components/footer/_macro-options.md:8]: Duplicate "logo." here

File: src/components/footer/_macro-options.md
Comment (rmccar) [inline src/components/footer/_macro-options.md:58]: Dont think we need to add "Helpful to add a margin utility class to control spacing." Its not the sort of thing we have put in the description of other params

File: src/components/footer/_macro-options.md
Comment (rmccar) [inline src/components/footer/_macro-options.md:63]: Think this should just be named logo to be consistent with other documentation. Ive suggested the same for the header

File: src/components/footer/_macro-options.md
Comment (alessioventuriniAND) [inline src/components/footer/_macro-options.md:68]: One missing >

**Review by alessioventuriniAND — COMMENTED** [review]
Can you let me know when you have fixed that issue with safari? 

After that I think I am happy to approve this PR considering that you have received approval from Dina around the margin and spacing of the logos.

**Review by alessioventuriniAND — APPROVED** [review]

File: src/components/footer/example-footer-with-multiple-logo.njk
Comment (rmccar) [inline src/components/footer/example-footer-with-multiple-logo.njk:1]: For consistency with the header examples this file should be named `src/components/footer/example-footer-with-multiple-logos.njk`

File: src/components/footer/example-footer-with-multiple-logo-and-split-display.njk
Comment (rmccar) [inline src/components/footer/example-footer-with-multiple-logo-and-split-display.njk:1]: `src/components/footer/example-footer-with-multiple-logos-and-split-display.njk` same with this file

File: src/components/footer/_macro-options.md
Comment (rmccar) [inline src/components/footer/_macro-options.md:59]:
```suggestion
| logos   | Object`<Logos>` | false    | Allows for up to two logos to be used in the footer.                                                                                                                                          |
```
This is an object not an array and should refer to logos not logo

**Review by rmccar — CHANGES_REQUESTED** [review]
Just a few minor things

**Review by SriHV — APPROVED** [review]

**Review by rmccar — APPROVED** [review]

**Review by rmccar — DISMISSED** [review]

File: src/components/footer/_footer.scss
Comment (rmccar) [inline src/components/footer/_footer.scss:40]: I think the names of these classes is confusing, these classes are called `_logo` and `__extra-logo` but these aren't actually set on the logos themselves. The css in `__logo` class isn't being applied to the logo either, this is setting the gap on the logo container. You could restructure this so that there was one class called `_logo-container` set the gap on that and then within that select the img or svg from there

**Review by rmccar — APPROVED** [review]

**Pull request merged**
